src/lib/autoinstall/script.rb

```diff
@@ -0,0 +1,413 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+require "shellwords"
+require "fileutils"
+
+require "transfer/file_from_url"
+
+Yast.import "AutoinstConfig"
+
+module Y2Autoinstallation
+  # Base class representing custom user script that runs during various phases of installation.
+  # User docs, rendered:
+  # https://documentation.suse.com/sles/15-SP1/html/SLES-all/configuration.html#createprofile-scripts
+  # source: https://github.com/SUSE/doc-sle/blob/master/xml/ay_custom_user_scripts.xml
+  class Script
+    include Yast::Logger
+    include Yast::Transfer::FileFromUrl
+    include Yast::I18n
+
+    # filename of script, used as ID. If it is empty then it is extracted from location or from
+    # file type.
+    # @see {#script_name} to get always non empty string for script name
+    # @return [String]
+    #
+    # @example of values
+    #   "test.sh"
+    #   "script.pl"
+    attr_reader :filename
+
+    # Source code of script. Either source or location should be defined.
+    # @return [String] empty string means not defined
+    attr_reader :source
+
+    # URI for location from which download script. Either source or location should be defined.
+    # @note supported URIs are all that Yast::Transfer::FileFromUrl supports + `relurl://`
+    # @return [String] empty string means not defined
+    attr_reader :location
+
+    # Flag if script runs in debug mode.
+    # @note works only for interpreter defined by keyword
+    # @return [Boolean]
+    attr_reader :debug
+
+    def initialize(hash)
+      @filename = hash["filename"] || ""
+      @source = hash["source"] || ""
+      @location = hash["location"] || ""
+      @debug = !!hash["debug"]
```

Comment:
This means the default is `false`, but the [previously documented](https://doc.opensuse.org/projects/autoyast/#id-1.9.5.34.11.3) default is `true`. Please fix or explain.

Reply:
Yes, intentional change as debug affect is smaller due to its work only with keywords. So I change defaut and also change it in documentation change. Also I found a bit confusing to have debug enabled by default.